dlc-oracle/src/main/scala/org/bitcoins/dlc/oracle/DLCOracle.scala

```diff
@@ -91,85 +91,207 @@ case class DLCOracle(private val extPrivateKey: ExtPrivateKeyHardened)(implicit
 
   def listPendingEventDbs(): Future[Vector[EventDb]] = eventDAO.getPendingEvents
 
-  def listEvents(): Future[Vector[Event]] = {
-    for {
-      rValDbs <- rValueDAO.findAll()
-      eventDbs <- eventDAO.findAll()
-      outcomes <- eventOutcomeDAO.findAll()
-    } yield {
-      val rValDbsByNonce = rValDbs.groupBy(_.nonce)
-      val outcomesByNonce = outcomes.groupBy(_.nonce)
-      eventDbs.map(db =>
-        Event(rValDbsByNonce(db.nonce).head, db, outcomesByNonce(db.nonce)))
+  def listEvents(): Future[Vector[OracleEvent]] = {
+    eventDAO.findAll().map { eventDbs =>
+      val events = eventDbs.groupBy(_.eventDescriptorTLV)
+      events.values.map(dbs => OracleEvent(dbs)).toVector
     }
   }
 
-  def getEvent(nonce: SchnorrNonce): Future[Option[Event]] = {
+  def getEvent(nonce: SchnorrNonce): Future[Option[OracleEvent]] = {
```

Comment:
nit: I prefer `findEvent()` for things like this as the return type is similar to the scala standard library find. I.e. 

```scala

val x: Option[Int] = Vector(1,2,3).find(_ == 1)

```

`get` is only used to return concrete `T` types, i.e. 

```scala

val x: Int = Vector(1,2,3).find(_ == 1).get

```

I guess at the end of the day I just care about being consistent if we want to use `get`